pkg/storage/driver/secrets.go

```diff
@@ -96,6 +96,13 @@ func (secrets *Secrets) List(filter func(*rspb.Release) bool) ([]*rspb.Release,
 			secrets.Log("list: failed to decode release: %v: %s", item, err)
 			continue
 		}
+		releases = append(releases, rls)
+	}
+
+	latestReleases := getLatestReleases(releases)
```

Comment:
This doesn't seem like it should be the responsibility of `List`. A call to `List` should be able to fetch more than just the latest releases. This should probably be refactored as a chain to the `filter` passed along to `List`. I'm going to take a look at alternatives to this approach.